store/pageiterator/iterator.go

```diff
@@ -0,0 +1,63 @@
+package pageiterator
```

Comment:
There is no need for this to be in its own package. Please add this logic to `store/types/iterator.go`. You will avoid any import cycles by having tests in their own package `package types_test`.

Reply:
thanks, i forgot about that option